packages/calcite-components/src/components/button/button.scss

```diff
@@ -404,12 +406,25 @@
 
     &:hover {
       --calcite-internal-button-shadow-color: var(--calcite-color-foreground-3);
+      stroke: var(--calcite-color-border-input);
     }
     &:focus {
       --calcite-internal-button-shadow-color: var(--calcite-color-foreground-3);
     }
     &:active {
       --calcite-internal-button-shadow-color: var(--calcite-color-foreground-3);
+      stroke: var(--calcite-color-text-3);
+    }
+  }
+}
+:host([appearance="solid"][kind="neutral"]) {
+  button,
+  a {
+    &:hover {
+      background-color: var(--calcite-color-border-3);
```

Comment:
Can you set the internal button CSS props? Some [props are already set up](https://github.com/Esri/calcite-design-system/blob/aPreciado88/9246-button-enhance-component-interactivity-states/packages/calcite-components/src/components/button/button.scss#L42-L53) to use them to help simplify styles.

Applies to other `background-color`, and possibly `border-color`, assignments.